### PR TITLE
Add StateService

### PR DESCRIPTION
**Pull request by ZhangTao1596**
Plugin to provide state service
 * StateRoot validation
 * StateRoot persist
 * MPT related RPC and Command.

### CODE REVIEW

File: src/MPTService/MPTService/config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+    "PluginConfiguration": {
+        "Path": "Data_MPT_{0}",
+        "FullState": false,
+        "Validators": [
+            "http://localhost:10332"
```
Comment (Tommo-L) [inline src/MPTService/MPTService/config.json:6]: Wait https://github.com/neo-project/neo/pull/2090

File: neo-modules.sln [hunk 2]
```diff
@@ -24,6 +24,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Neo.Network.RPC.Tests", "te
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Neo.Plugins.Storage.Tests", "tests\Neo.Plugins.Storage.Tests\Neo.Plugins.Storage.Tests.csproj", "{9E7EA895-302A-4C0C-BA9B-54F9A67AD75C}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MPTService", "src\MPTService\MPTService.csproj", "{940004B2-A76F-4C36-9FD0-7C936BD35826}"
```
Comment (Tommo-L) [inline neo-modules.sln:27]:
```suggestion
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "StateService", "src\MPTService\MPTService.csproj", "{940004B2-A76F-4C36-9FD0-7C936BD35826}"
```

Comment (ZhangTao1596) [reply neo-modules.sln]: Done.

**Comment by erikzhang** [comment]
I think we should move the classes in `Neo.Cryptography.MPT` from core to here.

File: src/StateService/StateService.csproj [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  <ItemGroup>
+    <PackageReference Include="Neo" Version="3.0.0-CI01036" />
+    <PackageReference Include="Neo.ConsoleService" Version="1.0.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\LevelDBStore\LevelDBStore.csproj" />
+    <ProjectReference Include="..\RpcServer\RpcServer.csproj" />
+    <ProjectReference Include="..\RpcClient\RpcClient.csproj" />
```
Comment (vncoelho) [inline src/StateService/StateService.csproj:21]: RpcClient as well? Why?

File: src/StateService/StatePlugin.RPC.cs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            }
+
+            var skey = key.AsSerializable<StorageKey>();
+            var sitem = MPTTrie<StorageKey, StorageItem>.VerifyProof(root_hash, skey, proof);
+            if (sitem is null) throw new RpcException(-100, "Verification failed");
+            return sitem.Value.ToHexString();
+        }
+
+        [RpcMethod]
+        public JObject VoteStateRoot(JArray _params)
```
Comment (vncoelho) [inline src/StateService/StatePlugin.RPC.cs:88]: I see now, the idea is to move from tasks in `ConsensusService` to tasks here on the MPTService done by `DesignedNodes`, right?

Comment (ZhangTao1596) [reply src/StateService/StatePlugin.RPC.cs:88]: Yes.

File: src/StateService/StateService.csproj (hunk 3 above)
Comment (ZhangTao1596) [reply src/StateService/StateService.csproj:21]: Send signature via `RPC`.

**Comment by erikzhang** [comment]
> Agree. But after we merge #1996.

Why? I think it's better to move neo-project/neo#1996 to `MPYService` too.

**Comment by ZhangTao1596** [comment]
> > Agree. But after we merge #1996.
> 
> Why? I think it's better to move [neo-project/neo#1996](https://github.com/neo-project/neo/pull/1996) to `MPYService` too.

I will move `MPT` to plugin first and re-commit [neo-project/neo#1996](https://github.com/neo-project/neo/pull/1996).

**Comment by ZhangTao1596** [comment]
Since we have a lot to change, I will close this and reopen another.

**Pull request closed without merging**